build.gradle

```diff
@@ -81,52 +59,19 @@ dependencies {
     testUtil sourceSets.test.output
 }
 
-def licenseFiles = copySpec {
-    //mockito license
-    from(".") { include 'LICENSE' }
-}
-
-task sourcesJar(type: Jar) {
-    jar {
-        baseName = 'mockito-core'
-        from(sourceSets.main.allSource)
-        with licenseFiles
+//need to declare dependencies before applying publishable java library
+//otherwise the pom file is incorrect
+apply from: "gradle/publishable-java-library.gradle"
```

Comment:
I really like this idea. I can emit a warning. I'll also check if we can work around using afterEvaluate for this use case.